[Infra UI] Update Host Disk Partition Average Metrics to 'max'

jennypavlova opened the pull request:
Closes [#179044](https://github.com/elastic/kibana/issues/179044)

## Summary
This PR changes the disk usage formula to `max` and removes the disk usage charts.

Charts removed: `Disk Usage`
- Hosts view 
- Asset Details Page/Flyout

Changes in KPIs / Table: 

| Before |  After | 
| ------- | ----- |
| <img width="1883" alt="image" src="https://github.com/elastic/kibana/assets/14139027/81735a00-857b-459b-bcd0-4e5c252a0265"> | <img width="1881" alt="image" src="https://github.com/elastic/kibana/assets/14139027/86d53a22-dc7b-40c4-bcbc-d4bf7b607a00"> |

- With Hosts Limit: 

<img width="1627" alt="image" src="https://github.com/elastic/kibana/assets/14139027/9ea4416a-fee0-49c1-b4f4-8df1d934a940">

## Testing:
- Go to Hosts view and check 
- In the table Disk usage column formula, tooltip and title are changed to `max`
- the Disk usage KPI (subtitle, tooltip & formula) is changed to `max` and verify that the Disk Usage chart is removed 
- Same steps (2 and 3) on Asset details flyout page: The Disk Usage by Mount Point Chart should be still there only the Disk Usage chart is removed 

https://github.com/elastic/kibana/assets/14139027/68107a8f-3d7f-4b68-a39a-bc2766f933b6

jennypavlova committed:
Change Disk usage formula to max and remove chanrts

jennypavlova commented on x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/components/kpis/kpi.tsx:
```diff
@@ -1,70 +0,0 @@
-/*
```
This file is a duplicate of [x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/components/kpis/kpi_charts.tsx](https://github.com/elastic/kibana/pull/180392/files#diff-23ccfc3412d969670195715f6ca5fcf6a3383a883e21d565d89532b96024662b) and it wasn't used anywhere so I removed it

jennypavlova committed:
Merge branch 'main' into 179044-infra-ui-update-host-disk-partition-average-metrics-to-max

jennypavlova commented:
/ci

jennypavlova committed:
Remove unused translations

jennypavlova commented:
/ci

jennypavlova committed:
Update tests

jennypavlova committed:
Merge branch 'main' into 179044-infra-ui-update-host-disk-partition-average-metrics-to-max

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

jennypavlova commented:
/ci

jennypavlova committed:
Update tests

crespocarlos reviewed (commented):
Thanks for the changes @jennypavlova. 

We also need to change the endpoint `/infra`. The query runs an avg on `system.filesystem.used.pct`, which is inconsistent with what the tooltip in the table column shows.

[Here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/metrics_data_access/common/inventory_models/host/metrics/snapshot/disk_space_usage.ts) is the aggregation that the query runs. 

The `/infra` endpoint accepts a `metrics` parameter, where the client can ask for specific metrics. I'm not sure if that parameter is necessary or if we could just manually build the list of metrics directly [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/server/routes/infra/lib/helpers/query.ts#L46)

jennypavlova commented on x-pack/plugins/metrics_data_access/common/inventory_models/host/metrics/charts/disk.ts:
```diff
@@ -66,7 +66,7 @@ const diskUsageByMountPoint: LensConfigWithId = {
       yAxis: [
         {
-          ...formulas.diskUsage,
+          ...formulas.diskUsageAverage,
```
No, I checked with Roshan in the issue and the `diskUsageByMountPoint` chart should remain unchanged.

jennypavlova committed:
CR changes

jennypavlova commented on x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_metrics_charts.test.ts:
```diff
@@ -119,6 +117,18 @@ describe('useKubernetesSectionMetricsCharts', () => {
+    expect(diskUsageChart).toHaveProperty('subtitle', 'Max');
+    expect(diskUsageChart).toHaveProperty('decimals', 1);
+  });
+
```
True, I will remove this one, thanks 👍

jennypavlova commented on x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_metrics_charts.ts:
```diff
@@ -118,12 +116,19 @@ export const useKubernetesSectionMetricsCharts = ({
   return charts;
 };
 
+const getSubtitleFromChartValue = (value: string) =>
```
Renamed ✅

jennypavlova committed:
Change to max

jennypavlova commented:
@crespocarlos Thank you for reviewing the PR!

> We also need to change the endpoint /infra. The query runs an avg on system.filesystem.used.pct, which is inconsistent with what the tooltip in the table column shows.
> 
> [Here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/metrics_data_access/common/inventory_models/host/metrics/snapshot/disk_space_usage.ts) is the aggregation that the query runs.

Good catch, thanks! I changed it to `max` there as well. I also renamed it to max inside [the table definition](https://github.com/elastic/kibana/pull/180392/files#diff-b8bd0aeb777d29980f7d07e50307b09b38d5d6fe3b736fa5dc8f4c44eaa879feR352)

> The /infra endpoint accepts a metrics parameter, where the client can ask for specific metrics. I'm not sure if that parameter is necessary or if we could just manually build the list of metrics directly [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/server/routes/infra/lib/helpers/query.ts#L46)

I am not sure I understand that completely. The current `diskSpaceUsage` inside the `InfraAssetMetricType` should be fine right (after the formula change in [disk_space_usage.ts](https://github.com/elastic/kibana/pull/180392/files#diff-8ca3226a89c5ab579b66ef8a431696509e4de61972cdab3754875989a68abe18R10)?

jennypavlova committed:
Merge branch 'main' into 179044-infra-ui-update-host-disk-partition-average-metrics-to-max

crespocarlos commented:
> I am not sure I understand that completely. The current `diskSpaceUsage` inside the `InfraAssetMetricType` should be fine right (after the formula change in [disk_space_usage.ts](https://github.com/elastic/kibana/pull/180392/files#diff-8ca3226a89c5ab579b66ef8a431696509e4de61972cdab3754875989a68abe18R10)?

Hey @jennypavlova , it should be fine. Let's just make sure this same metric isn't being used by the Inventory UI.

crespocarlos reviewed (approved):
LGTM!

jennypavlova committed:
Merge branch 'main' into 179044-infra-ui-update-host-disk-partition-average-metrics-to-max

jennypavlova commented:
> Hey @jennypavlova , it should be fine. Let's just make sure this same metric isn't being used by the Inventory UI.

I searched for it but couldn't find any other usage 👍

jennypavlova committed:
Update tests

jennypavlova committed:
Update tests

jennypavlova committed:
Merge branch 'main' into 179044-infra-ui-update-host-disk-partition-average-metrics-to-max
